### PR TITLE
Automated cherry pick of #2074: Using patch/apply on update pods.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
Cherry pick of #2074 on release-0.6.
#2074: Using patch/apply on update pods.
For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.
```release-note
Use PATCH updates for pods. This fixes support for Pods when using the latest features in Kubernetes v1.29
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
/release-note-edit

```release-note
Use PATCH updates for pods. This fixes support for Pods when using the latest features in Kubernetes v1.29
```

File: pkg/controller/jobs/pod/pod_controller_test.go [hunk 1]
```diff
@@ -3569,3 +3569,126 @@ func TestGetWorkloadNameForPod(t *testing.T) {
 		t.Errorf("Expected different workload name\n want: %s\n got: %s", wantWlName, wlName)
 	}
 }
+
+func TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps(t *testing.T) {
```
Comment (tenzen-y) [inline pkg/controller/jobs/pod/pod_controller_test.go:3573]: To avoid these conflicts, I will cherry-pick #1721 to release-0.6.

Comment (tenzen-y) [reply pkg/controller/jobs/pod/pod_controller_test.go:3573]: Uhm, the `TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps` could not be performed in the release-0.6 branch without this PR. (https://github.com/kubernetes-sigs/kueue/pull/2090)

So, could we just drop the `TestReconciler_DeletePodAfterTransientErrorsOnUpdateOrDeleteOps` from this cherry-pick PR?
@alculquicondor WDYT?

Comment (mbobrovskyi) [reply pkg/controller/jobs/pod/pod_controller_test.go]: Done

Comment (mbobrovskyi) [reply pkg/controller/jobs/pod/pod_controller_test.go]: @alculquicondor WDYT?

Comment (alculquicondor) [reply pkg/controller/jobs/pod/pod_controller_test.go]: I'm ok with removing the test.

/approve

@tenzen-y anything to add?

Comment (tenzen-y) [reply pkg/controller/jobs/pod/pod_controller_test.go]: Everything looks good to me.
Thanks!
/lgtm

**Pull request merged**